### PR TITLE
feat: [FFM-7942]: add archive button and modal into Flag Details Opti…

### PR DESCRIPTION
**Pull request by crystal-fong**
Adding the Archive button and Archive modal in a Flag's Detail page
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cf/components/FlagDetailsOptionsMenuButton/__tests__/FlagDetailsOptionsMenuButton.test.tsx [hunk 1]
```diff
@@ -18,9 +18,13 @@ import { FeatureIdentifier } from 'framework/featureStore/FeatureIdentifier'
 import type { FlagDetailsOptionsMenuButtonProps } from '../FlagDetailsOptionsMenuButton'
 import FlagDetailsOptionsMenuButton from '../FlagDetailsOptionsMenuButton'
 
-const renderComponent = (props: Partial<FlagDetailsOptionsMenuButtonProps> = {}): RenderResult => {
+const renderComponent = (
+  isArchivingFfOn = false,
```
Comment (laura-mannus) [inline src/modules/75-cf/components/FlagDetailsOptionsMenuButton/__tests__/FlagDetailsOptionsMenuButton.test.tsx:22]: its weird that FF is written like Ff here but I know that wasn't you

File: src/modules/75-cf/components/FlagDetailsOptionsMenuButton/__tests__/FlagDetailsOptionsMenuButton.test.tsx [hunk 2]
```diff
@@ -58,6 +62,19 @@ describe('FlagDetailsOptionsMenuButton', () => {
...
+    const isArchivingFfOn = false
+
+    test('it should render a DELETE and EDIT button when FFM_7127_FF_MFE_Onboarding_Detail is toggled OFF', async () => {
+      renderComponent(isArchivingFfOn)
+
+      userEvent.click(document.querySelector('[data-icon="Options"]') as HTMLButtonElement)
+      expect(document.querySelector('[data-icon="edit"]')).toBeInTheDocument()
+      expect(document.querySelector('[data-icon="trash"]')).toBeInTheDocument()
+
+      expect(screen.getAllByText('edit')[1]).toBeInTheDocument()
```
[the remark this answers is not in the document]
Comment (crystal-fong) [reply src/modules/75-cf/components/FlagDetailsOptionsMenuButton/__tests__/FlagDetailsOptionsMenuButton.test.tsx:74]: it's the edit icon and edit label. There's only one for delete bc it has a 'trash' icon and a 'delete' label

[the remark this answers is not in the document]
Comment (laura-mannus) [reply src/modules/75-cf/components/FlagDetailsOptionsMenuButton/__tests__/FlagDetailsOptionsMenuButton.test.tsx:74]: gotcha

**Review by knagurski — APPROVED** [review]
👨🏻‍🍳

**Pull request merged**